lib/git-lfs-s3/application.rb

```diff
@@ -74,6 +65,25 @@ def protected!
         body MultiJson.dump({message: 'Object not found'})
       end
     end
+    
+    def protected!
+      unless authorized?
+        response['WWW-Authenticate'] = %(Basic realm="Restricted Area")
+        throw(:halt, [401, "Invalid username or password"])
+      end
+    end
```

Comment:
Why move this method?  I generally advise against making structure changes like this since it makes the diff larger against upstream.